### PR TITLE
feat(Geosuggest): add onSuggestNoResults callback function

### PR DESCRIPTION
**Pull request by dimafeldman**
### Description

Added a callback function when there are no suggestions, to allow developers to handle this case easily, and show some UI indication for example.

### Checklist

- [x] All tests passing
- [x] Created tests which fail without the change (if possible)
- [x] Extended the README / documentation, if necessary
- [x] Commits and PR follow conventions

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -193,6 +193,12 @@ Gets triggered when a suggest is activated in the list. Only parameter is an obj
 * `label` – Type `String` – The label name
 * `placeId` – Type `String` – If it is a preset, equals the `label`. Else it is the Google Maps `placeID`
 
+#### onSuggestNoResults
+Type: `Function`
+Default: `function(value) {}`
```
Comment (ro-ka) [inline README.md]: There is no value as a param passed out.  :)

Comment (ro-ka) [inline README.md]: Or better name it `userInput` or `usedInput` or something like that.

File: src/suggest-list.jsx [hunk 2]
```diff
@@ -17,6 +17,14 @@ export default class SuggestList extends React.Component {
     return this.props.isHidden || this.props.suggests.length === 0;
   }
 
+  componentWillReceiveProps(newProps) {
+    if (newProps !== this.props) {
```
Comment (ro-ka) [inline src/suggest-list.jsx]: Isn’t a deeper comparison needed for these two objects? This just checks the reference…

**Comment by ro-ka** [comment]
🏁    Thanks for the PR! Would you mind to add some tests? @Caerostris wrote a lot to cover most parts of the lib. He surely can assist.

File: src/suggest-list.jsx (hunk 2 above)
Comment (dimafeldman) [inline src/suggest-list.jsx]: goot point, fixing

File: README.md (hunk 1 above)
Comment (dimafeldman) [inline README.md]: 👍

**Comment by dimafeldman** [comment]
Just did the adjustments + added a test to cover the addition

**Comment by ro-ka** [comment]
That’s great! Thank you!

**Comment by dimafeldman** [comment]
Just noticed that the dist files had not been updated, neither in npm. isn't it an automatic process?

**Comment by ro-ka** [comment]
We need to make a release, then the dist and npm and everything will be updated automatically. We’re close to releasing.

**Comment by dimafeldman** [comment]
👍

**Comment by ro-ka** [comment]
Just released in version `1.24.0`.

**Comment by dimafeldman** [comment]
Awesome!

**Pull request merged**